src/txmempool.cpp

```diff
@@ -120,47 +120,91 @@ void CTxMemPool::UpdateForDescendants(txiter updateIt, cacheMap& cachedDescendan
                                       const std::set<uint256>& setExclude, std::set<uint256>& descendants_to_remove,
                                       uint64_t ancestor_size_limit, uint64_t ancestor_count_limit)
 {
-    CTxMemPoolEntry::Children stageEntries, descendants;
-    stageEntries = updateIt->GetMemPoolChildrenConst();
-
-    while (!stageEntries.empty()) {
-        const CTxMemPoolEntry& descendant = *stageEntries.begin();
-        descendants.insert(descendant);
-        stageEntries.erase(descendant);
-        const CTxMemPoolEntry::Children& children = descendant.GetMemPoolChildrenConst();
-        for (const CTxMemPoolEntry& childEntry : children) {
-            cacheMap::iterator cacheIt = cachedDescendants.find(mapTx.iterator_to(childEntry));
-            if (cacheIt != cachedDescendants.end()) {
-                // We've already calculated this one, just add the entries for this set
-                // but don't traverse again.
-                for (txiter cacheEntry : cacheIt->second) {
-                    descendants.insert(*cacheEntry);
+    const CTxMemPoolEntry::Children& children = updateIt->GetMemPoolChildrenConst();
+
+    // initialize to hold all direct children
+    // children guaranteed to be unique at this point
+    std::vector<txiter> descendants(children.size());
+    size_t i = 0;
+    for (const auto& child: children) {
+        descendants[i] = mapTx.iterator_to(child);
+        ++i;
+    }
```

Comment:
i is needed to index descendants which is allocated up front.

it could be a std::transform I suppose, but I don't think std::transform provides a readability improvement here.

Reply:
> i is needed to index descendants which is allocated up front

Just `push_back`, and you don't need to allocate anything

Reply:
if i do a call to reserve instead of the sized constructor https://en.cppreference.com/w/cpp/container/vector/vector, that would work w/o extra allocations.

the sized constructor + default insert + indexed insert is a bit better IMO because there are some overheads to push_back/emplace_back in terms of updating the vec bookkeeping that the simple indexed insert above does not have, but it's all something that would need measuring as the compiler probably does an OK job at it.

do you have a strong preference around this?